webapp/src/ts/reducers/tasks.ts

```diff
@@ -0,0 +1,46 @@
+import { createReducer, on } from '@ngrx/store';
+
+import { Actions } from '@mm-actions/tasks';
+import { Actions as GlobalActions } from '@mm-actions/global';
+
+const initialState = {
+  tasksList: [],
+  selected: null,
+  loaded: false,
+};
+
+const orderBy = (t1, t2) => {
+  const lhs = t1?.dueDate;
+  const rhs = t2?.dueDate;
+  if (!lhs && !rhs) {
+    return 0;
+  }
+  if (!lhs) {
+    return 1;
+  }
+  if (!rhs) {
+    return -1;
+  }
+
+  return lhs < rhs ? -1 : 1;
+};
+
+const _tasksReducer = createReducer(
+  initialState,
+  on(GlobalActions.clearSelected, (state) => ({ ...state, selected: null })),
+
+  on(Actions.setTasksList, (state, { payload: { tasks } }) => {
+    return {
+      ...state,
+      tasksList: [...tasks].sort(orderBy),
```

Comment:
Maybe: rename orderBy to orderByDueDate to know right away how it's been sort.

Reply:
done!